src/lib/optimizer/optimization_context.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <memory>
+#include <utility>
+
+#include "cost_estimation/abstract_cost_estimator.hpp"
+
+namespace hyrise {
+
+/**
+ * OptimizationContext is used to track metadata about the optimization process / resulting LQP. Currently, it only
+ * tracks whether the resulting LQP is cacheable or not. If it is not cacheable, the SQLPipeline will not cache the
+ * optimized LQP.
+ */
+struct OptimizationContext {
+  explicit OptimizationContext(std::shared_ptr<AbstractCostEstimator> init_cost_estimator)
+      : cost_estimator{std::move(init_cost_estimator)} {}
```

Comment:
I guess we can remove the ctor and set the public member directly. I'm also not sure whether the shared ptr copy for the method call + move constructor is preferable over reference + copy ctor. We have to copy at some point anyways.